.github/workflows/workflow.yml

```diff
@@ -0,0 +1,50 @@
+name: Java CI
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: maven test and package
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Build with Maven
+      run: mvn -B package --file pom.xml
+
+  github_release:
+    if: startsWith(github.ref, 'refs/tags/v')
+    name: Create GitHub Release
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Build with Maven
+        run: mvn -B package --file pom.xml
```

Comment:
Currently, no tests are actually run, since I don't think maven can see the kinds of tests you made (and probably for good reason since some of them are pretty workload intensive). Might want to start adding some tests that can be ran automatically with `mvn test`

Reply:
Yeah some unit tests would be nice. I put up an issue on the project board for that - since the current tests like you said are very workload intensive, running as unit tests wouldn't be ideal. Also it's hard to test correctness with their output.